src/mapping/RadialGeoMultiscaleMapping.hpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <vector>
+#include "logging/Logger.hpp"
+#include "mapping/Mapping.hpp"
+
+namespace precice {
+namespace mapping {
+
+/// Geometric multiscale mapping in axial direction
+class RadialGeoMultiscaleMapping : public Mapping {
+public:
+  /**
+   * @brief Geometric multiscale nature of the mapping (spread or collect).
+   *
+   * A geometric multiscale mapping can either go from the 1D to the 2D/3D solver. Then, we call it "spread".
+   * Or from the 2D/3D to the 1D solver, which we call "collect".
+   */
+  enum MultiscaleType {
+    SPREAD,
+    COLLECT
+  };
+  enum MultiscaleAxis {
+    X,
+    Y,
+    Z
+  };
+
+  /**
+   * @brief Constructor.
+   *
+   * @param[in] constraint Specifies mapping to be consistent or conservative.
+   * @param[in] dimensions Dimensionality of the meshes
+   * @param[in] type Geometric multiscale type of the mapping
+   * @param[in] axis Main axis along which radial geometric multiscale coupling happens
+   */
+  RadialGeoMultiscaleMapping(Constraint constraint, int dimensions, MultiscaleType type, MultiscaleAxis axis);
+
+  /// Computes the mapping coefficients from the in- and output mesh.
```

Comment:
From the in- and output mesh? No comprendo.

Reply:
I updated the comment to `/// Computes the mapping coefficients needed to transfer data between the in- and output mesh.`, does that make more sense?
Maybe it should read `/// Takes care of compute-heavy operations needed only once to set up the mapping.`?

Reply:
This seems to be a copy-paste from `Mapping.hpp`, which also appears exactly in `NearestNeighborBaseMapping.hpp`, as well as some RBF hpp files.

Since we have not defined a mathematical form of this mapping, this does not make much sense here. The second comment (`/// Takes care of compute-heavy operations needed only once to set up the mapping.`) makes more sense to me in the current state, especially since it emphasizes the "only once".

Reply:
Incorporated the second comment.